File: 50-hard-tree-sequence-likelihood/python/main.py
import json
import sys


def solve(parent, sequences, prior, transition):
    pass


def main():
    obj = json.load(sys.stdin)
    result = solve(
        obj["parent"],
        obj["sequences"],
        obj["prior"],
        obj["transition"],
    )
    print(f"{result:.6f}")


if __name__ == "__main__":
    main()

File: 50-hard-tree-sequence-likelihood/go/main.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type input struct {
	Parent     []int       `json:"parent"`
	Sequences  []*string   `json:"sequences"`
	Prior      []float64   `json:"prior"`
	Transition [][]float64 `json:"transition"`
}

func solve(parent []int, sequences []*string, prior []float64, transition [][]float64) float64 {
	return 0
}

func main() {
	var in input
	if err := json.NewDecoder(os.Stdin).Decode(&in); err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
	fmt.Printf("%.6f\n", solve(in.Parent, in.Sequences, in.Prior, in.Transition))
}
